src/status.rs

```diff
@@ -539,6 +553,131 @@ fn filter_block_txs<T: Send>(
         .into_iter()
 }
 
+pub(crate) struct OutPointStatus {
+    outpoint: OutPoint,
+    funding: Option<Height>,
+    spending: Option<(Txid, Height)>,
+    tip: BlockHash,
+}
+
+impl Serialize for OutPointStatus {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map = serializer.serialize_map(None)?;
+        if let Some(funding) = &self.funding {
+            map.serialize_entry("height", &funding)?;
+        }
+        if let Some((txid, height)) = &self.spending {
+            map.serialize_entry("spender_txhash", &txid)?;
+            map.serialize_entry("spender_height", &height)?;
+        }
+        map.end()
+    }
+}
+
+impl OutPointStatus {
+    pub(crate) fn new(outpoint: OutPoint) -> Self {
+        Self {
+            outpoint,
+            funding: None,
+            spending: None,
+            tip: BlockHash::all_zeros(),
+        }
+    }
+
+    pub(crate) fn sync(
+        &mut self,
+        index: &Index,
+        mempool: &Mempool,
+        daemon: &Daemon,
+    ) -> Result<bool> {
+        let funding = self.sync_funding(index, daemon, mempool)?;
+        let spending = self.sync_spending(index, daemon, mempool)?;
+        let same_status = (self.funding == funding) && (self.spending == spending);
+        self.funding = funding;
+        self.spending = spending;
+        self.tip = index.chain().tip();
+        Ok(!same_status)
+    }
+
+    /// Return true iff current tip became unconfirmed
+    fn is_reorg(&self, chain: &Chain) -> bool {
+        chain.get_block_height(&self.tip).is_none()
+    }
+
+    fn sync_funding(
+        &self,
+        index: &Index,
+        daemon: &Daemon,
+        mempool: &Mempool,
+    ) -> Result<Option<Height>> {
+        let chain = index.chain();
+        if !self.is_reorg(chain) {
+            if let Some(Height::Confirmed { .. }) = &self.funding {
+                return Ok(self.funding);
+            }
+        }
+        let mut confirmed = None;
+        daemon.for_blocks(
+            index.filter_by_txid(self.outpoint.txid),
+            |blockhash, block| {
+                if confirmed.is_none() {
+                    for tx in block.txdata {
+                        let txid = tx.txid();
+                        let output_len = u32::try_from(tx.output.len()).unwrap();
+                        if self.outpoint.txid == txid && self.outpoint.vout < output_len {
```

Comment:
> `self.outpoint.vout < output_len`

maybe i am missing something, but for what reason is this check here?

Reply:
Clearly this avoids out-of-bounds access.

Reply:
hmm i tried to find where `self.outpoint.vout` is used to index `tx.output`, but could not find it, `self.outpoint.vout` is not used after this here, which made sense after i thought of this: this code checks for output _funding_, it doesnt matter which output index is subscribed to, as long as it is on the transaction (`self.outpoint.vout < output_len`)

Reply:
It's indeed a sanity check - in case the user tries to subscribe to a non-existing outpoint (by specifying a `vout` larger than the actual # of `txid` outputs).

Reply:
What I'm trying to say is that it now *silently* ignores the transaction instead of printing a warning and a warning would be helpful because doing this is a bug and without a message it could be hard to analyze.